File: ArchitectureComponents/Room/PersonContact/app/src/main/java/com/dodemy/android/personcontact/database/PersonDao.java
package com.dodemy.android.personcontact.database;


import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.dodemy.android.personcontact.model.Person;

import java.util.List;

@Dao
public interface PersonDao {

    @Query("SELECT * FROM PERSON ORDER BY ID")
    List<Person> loadAllPersons();

    @Insert
    void insertPerson(Person person);

    @Update
    void updatePerson(Person person);

    @Delete
    void delete(Person person);

    @Query("SELECT * FROM PERSON WHERE id = :id")
    Person loadPersonById(int id);
}

File: Kotlin/PersonInfoRoom/app/src/main/java/com/dodemy/personinforoom/database/PersonDao.kt
package com.dodemy.personinforoom.database

import androidx.room.*
import com.dodemy.personinforoom.model.Person


@Dao
interface PersonDao {
    @Query("SELECT * FROM PERSON ORDER BY ID")
    fun loadAllPersons(): List<Person?>?

    @Insert
    fun insertPerson(person: Person?)

    @Update
    fun updatePerson(person: Person?)

    @Delete
    fun delete(person: Person?)

    @Query("SELECT * FROM PERSON WHERE id = :id")
    fun loadPersonById(id: Int): Person?
}
